src/engine/server/scaling_handler.go

```diff
@@ -0,0 +1,117 @@
+package server
+
+import (
+	"cf"
+	"db"
+	"models"
+
+	"code.cloudfoundry.org/cfhttp/handlers"
+	"code.cloudfoundry.org/lager"
+
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+const TokenTypeBearer = "bearer"
+
+type ScalingHandler struct {
+	cfClient cf.CfClient
+	logger   lager.Logger
+	database db.PolicyDB
+}
+
+func NewScalingHandler(logger lager.Logger, cfc cf.CfClient, database db.PolicyDB) *ScalingHandler {
+	return &ScalingHandler{
+		cfClient: cfc,
+		logger:   logger,
+		database: database,
+	}
+}
+
+func (h *ScalingHandler) HandleScale(w http.ResponseWriter, r *http.Request) {
+	trigger := &models.Trigger{}
+	err := json.NewDecoder(r.Body).Decode(trigger)
+	if err != nil {
+		h.logger.Error("handle-scale-unmarshal-trigger", err)
+		handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
+			Code:    "Bad-Request",
+			Message: "Incorrect trigger in request body"})
+		return
+	}
+
+	h.logger.Debug("handle-scale", lager.Data{"trigger": trigger})
+
+	var newInstances int
+	newInstances, err = h.Scale(trigger)
+	if err != nil {
+		h.logger.Error("handle-scale-perform-scaling-action", err, lager.Data{"trigger": trigger})
+		handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+			Code:    "Internal-server-error",
+			Message: "Error taking scaling action"})
+		return
+	}
+	handlers.WriteJSONResponse(w, http.StatusCreated, models.AppProperties{Instances: newInstances})
```

Comment:
201 is used to indicate that "a scaling action has been created".  Since we don't observe whether it is actually scaled to the number that we expect.

and CF API PUT /v2/apps/:guid returns "201 created"